src/main/java/uk/gov/hmcts/reform/divorce/orchestration/service/impl/BulkCaseServiceImpl.java

```
@@ -171,4 +178,37 @@ public void handleBulkCaseAcceptedCasesEvent(BulkCaseAcceptedCasesEvent event) {
         final long endTime = Instant.now().toEpochMilli();
         log.info("Completed bulk case removed link from cases with bulk case Id:{} in:{} millis", bulkCaseId, endTime - startTime);
     }
+
+    @Override
+    @EventListener
+    public void handleBulkCaseRemovePronouncementDetailsEvent(BulkCaseRemovePronouncementDetailsEvent event) {
+        final long startTime = Instant.now().toEpochMilli();
+
+        TaskContext context = (TaskContext) event.getSource();
+        CaseDetails caseResponse = event.getCaseDetails();
+        List<String> casesToUnlink = context.getTransientObject(REMOVED_CASE_LIST);
+        final String bulkCaseId = caseResponse.getCaseId();
+
+        for (String caseId : casesToUnlink) {
+            try {
+                // Triggering event does not require payload
+                updateCaseWorkflow.run(
+                        Collections.emptyMap(),
+                        context.getTransientObject(AUTH_TOKEN_JSON_KEY),
+                        caseId,
+                        REMOVE_FROM_BULK_CASE_LISTED_EVENT
+                );
+            } catch (WorkflowException e) {
+                log.error("Error removing bulk case pronouncement details with bulkCaseId: {} and caseId {}", bulkCaseId, caseId);
```

Worth logging the exception? Also do we want to silently continue if this happens?

---

Because the below info log will print even in a failure

---

Yes thats fine, this works on all individual cases in a bulk, so if one fails we won't stop the event triggering on the remaining cases. Because this is on an event, the callback has already finished so failing here does nothing to stop the bulk case event. Instead its up to the caseworkers / support to check the failing cases I believe.